From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: granny <granny@purpurmc.org>
Date: Thu, 16 May 2024 20:37:05 -0700
Subject: [PATCH] Rebrand


diff --git a/src/main/java/io/papermc/paper/ServerBuildInfo.java b/src/main/java/io/papermc/paper/ServerBuildInfo.java
index 7107eabf02d9d9b69d97c978ba421ad52031ec3f..7369179838293141b1fb981cc762e3ffac18e3c3 100644
--- a/src/main/java/io/papermc/paper/ServerBuildInfo.java
+++ b/src/main/java/io/papermc/paper/ServerBuildInfo.java
@@ -25,6 +25,13 @@ public interface ServerBuildInfo {
     Key BRAND_PURPUR_ID = Key.key("purpurmc", "purpur");
     // Purpur end
 
+    // StevenPur start
+    /**
+     * The brand id for StevenPur.
+     */
+    Key BRAND_STEVENPUR_ID = Key.key("stevenpurmc", "stevenpur");
+    // StevenPur end
+
     /**
      * Gets the {@code ServerBuildInfo}.
      *
