From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: qwqtoday <lukas.leung@hotmail.com>
Date: Wed, 5 Jun 2024 15:20:16 +0800
Subject: [PATCH] Make nether world scale 1


diff --git a/src/main/java/net/minecraft/world/entity/Entity.java b/src/main/java/net/minecraft/world/entity/Entity.java
index 7c8a3a6af8d5fdbb2341fe4012686bfc4f79a312..5c56dfd9970bfdd14a2c20c030a1f4608b9a6214 100644
--- a/src/main/java/net/minecraft/world/entity/Entity.java
+++ b/src/main/java/net/minecraft/world/entity/Entity.java
@@ -3858,7 +3858,7 @@ public abstract class Entity implements SyncedDataHolder, Nameable, EntityAccess
                 return null;
             } else {
                 WorldBorder worldborder = destination.getWorldBorder();
-                double d0 = DimensionType.getTeleportationScale(this.level().dimensionType(), destination.dimensionType());
+                double d0 = 1.0D;
                 BlockPos blockposition = worldborder.clampToBounds(this.getX() * d0, this.getY(), this.getZ() * d0);
                 // CraftBukkit start
                 // Paper start - Configurable portal search radius
